File: flutter_app/lib/features/model/appeal_review.dart
class AppealReviewModel {
  final int? reviewId;
  final int? appealId;
  final String? reviewLevel;
  final DateTime? reviewTime;
  final String? reviewer;
  final String? reviewerDept;
  final String? reviewResult;
  final String? reviewOpinion;
  final String? suggestedAction;
  final double? suggestedFineAmount;
  final int? suggestedPoints;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const AppealReviewModel({
    this.reviewId,
    this.appealId,
    this.reviewLevel,
    this.reviewTime,
    this.reviewer,
    this.reviewerDept,
    this.reviewResult,
    this.reviewOpinion,
    this.suggestedAction,
    this.suggestedFineAmount,
    this.suggestedPoints,
    this.createdAt,
    this.updatedAt,
  });

  factory AppealReviewModel.fromJson(Map<String, dynamic> json) {
    return AppealReviewModel(
      reviewId: json['reviewId'],
      appealId: json['appealId'],
      reviewLevel: json['reviewLevel'],
      reviewTime:
          json['reviewTime'] != null ? DateTime.tryParse(json['reviewTime']) : null,
      reviewer: json['reviewer'],
      reviewerDept: json['reviewerDept'],
      reviewResult: json['reviewResult'],
      reviewOpinion: json['reviewOpinion'],
      suggestedAction: json['suggestedAction'],
      suggestedFineAmount: (json['suggestedFineAmount'] as num?)?.toDouble(),
      suggestedPoints: json['suggestedPoints'],
      createdAt:
          json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
      updatedAt:
          json['updatedAt'] != null ? DateTime.tryParse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'reviewId': reviewId,
        'appealId': appealId,
        'reviewLevel': reviewLevel,
        'reviewTime': reviewTime?.toIso8601String(),
        'reviewer': reviewer,
        'reviewerDept': reviewerDept,
        'reviewResult': reviewResult,
        'reviewOpinion': reviewOpinion,
        'suggestedAction': suggestedAction,
        'suggestedFineAmount': suggestedFineAmount,
        'suggestedPoints': suggestedPoints,
        'createdAt': createdAt?.toIso8601String(),
        'updatedAt': updatedAt?.toIso8601String(),
      };
}

File: backend/src/main/java/com/tutict/finalassignmentbackend/entity/AppealReview.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 申诉审核表实体类，对应数据库表 "appeal_review"
 * 记录申诉的审核信息
 */
@Data
@TableName("appeal_review")
public class AppealReview implements Serializable {
    /**
     * 序列化版本 UID
     */
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 审核记录 ID，主键，自动增长
     */
    @TableId(value = "review_id", type = IdType.AUTO)
    private Long reviewId;

    /**
     * 申诉记录 ID
     */
    @TableField("appeal_id")
    private Long appealId;

    /**
     * 审核级别 (Primary, Secondary, Final)
     */
    @TableField("review_level")
    private String reviewLevel;

    /**
     * 审核时间
     */
    @TableField("review_time")
    private LocalDateTime reviewTime;

    /**
     * 审核人
     */
    @TableField("reviewer")
    private String reviewer;

    /**
     * 审核部门
     */
    @TableField("reviewer_dept")
    private String reviewerDept;

    /**
     * 审核结果 (Approved, Rejected, Need_Resubmit, Transfer)
     */
    @TableField("review_result")
    private String reviewResult;

    /**
     * 审核意见
     */
    @TableField("review_opinion")
    private String reviewOpinion;

    /**
     * 处理建议 (Cancel_Offense, Reduce_Fine, Reduce_Points, Reject_Appeal, Other)
     */
    @TableField("suggested_action")
    private String suggestedAction;

    /**
     * 建议罚款金额 (元)
     */
    @TableField("suggested_fine_amount")
    private BigDecimal suggestedFineAmount;

    /**
     * 建议扣分
     */
    @TableField("suggested_points")
    private Integer suggestedPoints;

    /**
     * 创建时间
     */
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;

    /**
     * 备注
     */
    @TableField("remarks")
    private String remarks;
}